src/HeatmapLayer.js

```diff
@@ -74,6 +78,76 @@ function shouldIgnoreLocation(loc: LngLat): boolean {
   return isInvalid(loc.lng) || isInvalid(loc.lat);
 }
 
+export function computeAggregate(
+  agg: Object,
+  intensity: number,
+  aggregateType: AggregateType = 'sum'
+): number {
+  /* eslint-disable no-use-before-define */
+  const updateMeanAndSigma = (m, c, v) => {
+    const newMean = agg.data.mean + fns.mean(agg.data.mean, c, v);
+
+    agg.data.sigma += (v - newMean) * (v - agg.data.mean);
+
+    agg.data.mean = newMean;
+  };
+
+  /* eslint-disable no-unused-vars */
+  const fns = {
+    mean: (m, c, v) => (v - m) / c,
+    count: (m, c, v) => 1,
+    sum: (m, c, v) => v,
+    distinct: (m, c, v) => {
+      agg.same.push(v);
```

Comment:
It'd be more efficient if you used the [es6 `Set`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set), since that'll also do your uniqueness, in a more efficient way.

IE
```javascript
agg.same.add(v);
return agg.same.length;
```

Also has the nice advantage that you don't need `uniq` anymore.